arkouda/pdarraysetops.py

```diff
@@ -309,47 +311,76 @@ def union1d(pda1: pdarray, pda2: pdarray) -> pdarray:
     -----
     ak.union1d is not supported for bool or float64 pdarrays
 
+    Both inputs must match types in that they must both be pdarrays or both be iterable
+
     Examples
     --------
+    Singular pdarrays
     >>> ak.union1d(ak.array([-1, 0, 1]), ak.array([-2, 0, 2]))
     array([-2, -1, 0, 1, 2])
+
+    >>> ak.union1d((ak.array([0, 2]), ak.array([-1, 0, 1])), (ak.array([0, 2]), ak.array([-2, 0, 2])))
+    [-2, -1, 0]
+    [1, 2]
+    (array(0, 3), array([-2, -1, 0, 1, 2])
+
+    Multiple pdarrays flattened
+    >>> a_segs = ak.array([0, 3])
+    >>> a_vals = ak.array([0, 1, 2, 3, 4])
+    >>> b_segs = ak.array([0, 1])
+    >>> b_vals = ak.array([5, 6, 7, 8])
+    >>> ak.union1d([a_segs, a_vals], [b_segs, b_vals])
+    (array([0, 4]), array([0, 1, 2, 3, 4, 5, 6, 7, 8]))
     """
-    if pda1.size == 0:
-        return pda2  # union is pda2
-    if pda2.size == 0:
-        return pda1  # union is pda1
-    if pda1.dtype == int and pda2.dtype == int:
-        repMsg = generic_msg(cmd="union1d", args="{} {}". \
-                             format(pda1.name, pda2.name))
-        return cast(pdarray, create_pdarray(repMsg))
-    return cast(pdarray,
-                unique(cast(pdarray,
-                            concatenate((unique(pda1), unique(pda2)), ordered=False))))  # type: ignore
+    if isinstance(pda1, pdarray) or isinstance(pda2, pdarray):
+        if not (isinstance(pda1, pdarray) and isinstance(pda2, pdarray)):
```

Comment:
Could this check be changed to just a single `if isinstance(pda1, pdarray) and isinstance(pda2, pdarray)`? It seems equivalent to me, but I'm not certain

Reply:
I see, thanks for clarifying.